### PR TITLE
fix(medusa): Pricing update fixes

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- update unit price for line-items added with `should_merge: true` where combined quantity warrants an updated price
- update-line-item should ignore calculated price if item is a draft order

### CODE REVIEW

File: packages/medusa/src/services/cart.ts [hunk 1]
```diff
@@ -900,10 +901,36 @@ class CartService extends TransactionBaseService {
...
+              variantsPricing[currentItem.variant_id!]
+
             lineItemsToUpdate[currentItem.id] = {
               quantity: item.quantity,
               has_shipping: false,
             }
+
+            if (
+              isDefined(calculated_price) &&
+              currentItem.unit_price !== calculated_price
```
Comment (adrien2p) [inline packages/medusa/src/services/cart.ts]: todo: I think we can assign the calculated prices even though it is the same, it shouldn't be a problem and we can remove the second condition. wdyt?

File: packages/medusa/src/services/cart.ts [hunk 2]
```diff
@@ -1021,24 +1049,26 @@ class CartService extends TransactionBaseService {
               )
             }
 
-            const variantsPricing = await this.pricingService_
-              .withTransaction(transactionManager)
-              .getProductVariantsPricing(
-                [
+            if (should_calculate_prices) {
```
Comment (adrien2p) [inline packages/medusa/src/services/cart.ts:1049]: question: shouldn't we always calculate the prices when we update the line item?

**Review by adrien2p — APPROVED** [review]
LGTM, few comments

File: packages/medusa/src/services/cart.ts (hunk 2 above)
Comment (pKorsholm) [reply packages/medusa/src/services/cart.ts:1049]: no, this was how it was handled previously which makes sense from a store perspective, but breaks custom unit prices set from the draft orders (context: #5274) 😄 made the same mistake in the initial pr introducing this calculation

File: packages/medusa/src/services/cart.ts (hunk 1 above)
Comment (pKorsholm) [reply packages/medusa/src/services/cart.ts]: sure, doesn't make a difference 😄

File: packages/medusa/src/services/cart.ts (hunk 2 above)
Comment (adrien2p) [reply packages/medusa/src/services/cart.ts:1049]: ahah got it 👍 thanks for the context

**Pull request merged**